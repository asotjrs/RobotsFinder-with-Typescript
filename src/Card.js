import React from "react";

const Card=({input})=>{

    return <div className={" bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5 tc"}>
        <img src={'https://robohash.org/'+input} alt={'robot'}/>
        <div>
            <h2>{input}</h2>
            <p>{input+'@gmail.com'}</p>
        </div>
    </div>

};
export default Card;