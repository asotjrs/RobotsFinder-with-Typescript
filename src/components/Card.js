import React from "react";

const Card=({robot})=>{
    return robot===""?null:
     <div className={" bg-light-red dib br3 pa3 ma2 grow bw2 shadow-5 tc"}>
        <img src={'https://robohash.org/'+robot+"?size=200x200"} alt={'robot'}/>
        <div>
            <h2>{robot}</h2>
            <p>{robot.replace(' ','_')+'@gmail.com'}</p>
        </div>
    </div>

};
export default Card;