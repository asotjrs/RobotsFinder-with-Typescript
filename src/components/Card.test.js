import {shallow} from 'enzyme';
import React from "react";
import Card from "./Card";
import toJson from "enzyme-to-json";
it('to match snap-shot', function () {
    const mockrobot="just testing";
    expect(toJson(shallow(<Card robot={mockrobot}/>))).toMatchSnapshot();

});


