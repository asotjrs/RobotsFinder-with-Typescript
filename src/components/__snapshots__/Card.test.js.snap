// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to match snap-shot 1`] = `
<div
  className=" bg-light-red dib br3 pa3 ma2 grow bw2 shadow-5 tc"
>
  <img
    alt="robot"
    src="https://robohash.org/just testing"
  />
  <div>
    <h2>
      just testing
    </h2>
    <p>
      just_testing@gmail.com
    </p>
  </div>
</div>
`;
